bundles/automation/org.eclipse.smarthome.automation.provider.file/src/main/java/org/eclipse/smarthome/automation/internal/provider/file/AbstractFileProvider.java

```diff
@@ -256,9 +256,19 @@ protected void importFile(String parserType, URL url) {
 
     protected void updateProvidedObjectsHolder(URL url, Set<E> providedObjects) {
         if (providedObjects != null && !providedObjects.isEmpty()) {
+            List<String> previousPortfolio = providerPortfolio.get(url);
             List<String> uids = new ArrayList<String>();
             for (E providedObject : providedObjects) {
                 String uid = getUID(providedObject);
+                if (providedObjectsHolder.get(uid) == null) {
+                    if (checkExistence(uid)) {
+                        continue;
+                    }
+                } else if (previousPortfolio == null || !previousPortfolio.contains(uid)) {
```

Comment:
Either this is a bug or the code is hard to read - wouldn't you have a problem IF the previousPortfolio already contains this uid?